programs/drift/src/controller/orders.rs

```diff
@@ -3228,6 +3244,43 @@ pub fn fulfill_spot_order_with_match(
         return Ok(0_u64);
     }
 
+    let (taker_max_base_asset_amount, taker_max_quote_asset_amount) =
+        get_max_fill_amounts(taker, taker_order_index, base_market, quote_market)?;
+
+    let taker_implied_max_base_asset_amount =
+        if let Some(taker_max_quote_asset_amount) = taker_max_quote_asset_amount {
+            standardize_base_asset_amount(
+                taker_max_quote_asset_amount
+                    .safe_mul(base_market.get_precision())?
+                    .safe_div(maker_price)?,
+                base_market.order_step_size,
+            )?
+        } else {
+            u64::MAX
+        };
+    let taker_base_asset_amount = taker_base_asset_amount
```

Comment:
probs easier to understand with something like ? (same for maker part)
```
taker_base_asset_amount  = if let Some(taker_max_quote_asset_amount) = taker_max_quote_asset_amount {
    // ...
    taker_base_asset_amount.min(taker_implied_max_base_asset_amount);
} else { 
  taker_base_asset_amount.min(taker_max_base_asset_amount.unwrap());
}
```

Reply:
sure